src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php

```diff
@@ -32,14 +32,47 @@ public function configureOptions(OptionsResolver $resolver): void
 
                 $choiceTranslationLocale = $options['choice_translation_locale'];
 
-                return ChoiceList::loader($this, new IntlCallbackChoiceLoader(static fn () => array_flip(Currencies::getNames($choiceTranslationLocale))), $choiceTranslationLocale);
+                return ChoiceList::loader(
+                    $this,
+                    new IntlCallbackChoiceLoader(
+                        static function () use ($options, $choiceTranslationLocale) {
+                            $active = $options['active'];
+
+                            $legalTender = $options['legal_tender'];
+
+                            // When the `active` and `legal_tender` options are null, it is unnecessary no need to retrieve the currencies from the meta indice.
+                            if (null === $active && null === $legalTender) {
+                                return array_flip(Currencies::getNames($choiceTranslationLocale));
+                            }
+
+                            $filteredCurrencyNames = [];
+
+                            foreach (Currencies::getCurrencyCodes() as $code) {
+                                if (!Currencies::isValidInAnyCountry($code, $legalTender, $active, $options['active_or_not_at'])) {
+                                    continue;
+                                }
+
+                                $filteredCurrencyNames[$code] = Currencies::getName($code, $choiceTranslationLocale);
+                            }
+
+                            return array_flip($filteredCurrencyNames);
+                        },
+                    ),
+                    $choiceTranslationLocale
+                );
             },
             'choice_translation_domain' => false,
             'choice_translation_locale' => null,
+            'active' => true,
+            'active_or_not_at' => new \DateTimeImmutable('today', new \DateTimeZone('Etc/UTC')),
```

Comment:
what about this? (with a check to ensure only one of them is set at a time? I don't like the name `active_or_not_at` ...

```suggestion
            'active_at' => new \DateTimeImmutable(),
            'not_active_at' => null,
```

Reply:
As long as the two parameters are not set together, this could work yes. 👍🏻

Reply:
I wouldn't use `_at` suffix because that means "active at that specific moment of time" (not before or after).

I think we need to use `_before` or `_after` suffix depending on the option: `active_before`

Reply:
Done 👍🏻